non-boot clusters nil pointer on `jx step helm apply`

Note that this isn't in the version stream - it's caused by #6771, and is in fact causing the version stream update to fail.

The staging (and production) env pipelines in a `jx install` cluster with Tekton (not with static masters - we caught that problem before #6771 was merged, but due to #6811 we didn't catch the Tekton case) will fail with an error like:

```
Showing logs for build jenkins-x-versions-bot-test-env-hj4rl-1 stage from-build-pack and container step-build-helm-apply
Modified file /workspace/source/env/Chart.yaml to set the chart to version 1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x248 pc=0x26dafd3]

goroutine 1 [running]:
github.com/jenkins-x/**/pkg/cmd/step/helm.(*StepHelmApplyOptions).Run(0xc00083b290, 0x0, 0x0)
	/workspace/source/pkg/cmd/step/helm/step_helm_apply.go:255 +0xde3
github.com/jenkins-x/**/pkg/cmd/step/helm.NewCmdStepHelmApply.func1(0xc0009d4c80, 0xa733d88, 0x0, 0x0)
	/workspace/source/pkg/cmd/step/helm/step_helm_apply.go:85 +0x6f
github.com/spf13/cobra.(*Command).execute(0xc0009d4c80, 0xa733d88, 0x0, 0x0, 0xc0009d4c80, 0xa733d88)
	/workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x2ae
github.com/spf13/cobra.(*Command).ExecuteC(0xc000493680, 0xc0005bfc00, 0x5ace800, 0xc0000b8000)
	/workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fc
github.com/spf13/cobra.(*Command).Execute(...)
	/workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
github.com/jenkins-x/**/cmd/**/app.Run(0x0, 0x0, 0x0, 0xc0008a3f88, 0xc00009e058)
	/workspace/source/cmd/**/app/**.go:19 +0x185
main.main()
	/workspace/source/cmd/**/**.go:11 +0x32
```

This is because we end up getting a nil `requirements`. We special-case static masters currently, but we really need to handle all cases where we're on a `jx install` cluster the same in this regard.

--- a/pkg/cmd/step/helm/step_helm_apply.go
+++ b/pkg/cmd/step/helm/step_helm_apply.go
@@ -406,18 +406,17 @@ func (o *StepHelmApplyOptions) getRequirements() (*config.RequirementsConfig, st
 		return nil, "", errors.Wrap(err, "getting the team setting from the cluster")
 	}
 
-	// TODO To stay backwards compatible with code after fixing #6653, we need to create a dummy RequirementsConfig here
-	// TODO This is a special case for static master and needs to be removed (HF)
-	if !teamSettings.IsJenkinsXPipelines() {
+	requirements, err = config.GetRequirementsConfigFromTeamSettings(teamSettings)
+	if err != nil {
+		return nil, "", errors.Wrap(err, "getting the requirements from team settings")
+	}
+	// TODO: Workaround for non-boot clusters. Remove when we get rid of jx install. (APB)
+	if requirements == nil {
 		requirements = config.NewRequirementsConfig()
 		requirementsFileName = config.RequirementsConfigFileName
 		return requirements, requirementsFileName, nil
 	}
 
-	requirements, err = config.GetRequirementsConfigFromTeamSettings(teamSettings)
-	if err != nil {
-		return nil, "", errors.Wrap(err, "getting the requirements from team settings")
-	}
 	return requirements, "", nil
 }
 

